### ISSUE TITLE
Can't remove or disable Title field

### ISSUE THREAD

**Issue by jonom** [dnadesign/silverstripe-elemental#632]
I want to have a fairly simple Photo block which just allows you to choose an image and set a couple of options. Setting a title on this is redundant and unnecessary work for the user because the title of the selected photo can be used automatically.

But trying to remove the Title throws an error, and disabling it doesn't work either :(

(BTW I'm just trying this module out for the first time - it looks pretty nice overall!)

```php
use DNADesign\Elemental\Models\BaseElement;
use SilverStripe\Assets\Image;

class PhotoElement extends BaseElement
{
    private static $has_one = [
        'Photo' => Image::class,
    ];

    public function getCMSFields()
    {
        $fields = parent::getCMSFields();

        // This removes the text field nested within the TextCheckboxGroupField and throws Uncaught Error: Call to a member function getName() on null IN GET /admin/elemental-area/schema/elementForm/1 Line 60 in (...) vendor/dnadesign/silverstripe-elemental/src/Forms/EditFormFactory.php
        $fields->removeByName('Title');
        // This doesn't do anything
        $fields->dataFieldByName('Title')->performReadonlyTransformation();

        // Target the TextCheckboxGroupField combo field (it's name is Title but so is the text field within it)
        $titleComboField = $fields->fieldByName('Title'); // No result - doesn't work
        $titleComboField = $fields->dataFieldByName('Title'); // Wrong - returns the text field nested within the TextCheckboxGroupField
        $titleComboField = $fields->findTab('Root.Main')->Fields()->first(); // This does the trick but it's fragile and could break very easily

        // Try to delete the whole field (doesn't work)
        $fields->remove($titleComboField);

        // Try to make the whole field read only at least (doesn't work)
        $titleComboField->performReadonlyTransformation();

        return $fields;
    }

    public function getTitle()
    {
        return $this->Photo()->getTitle();
    }
}
```

**Comment by ScopeyNZ** [comment]
I've raised a PR at #642 that I've marked as "resolving" this issue. With it just a simple `->removeByName('Title')` will now work. 

Replacing is another thing entirely.

**Comment by robbieaverill** [comment]
Fixed with #642. Please reopen if I've missed something!

**Comment by jonom** [comment]
Much appreciated guys!
